### PR TITLE
Code cleanup for LocalEvent class

### PR DESCRIPTION
**Pull request by ihhub**
- add error handling code for SDL functions
- rename majority of class' methods
- make enum class over just normal enum
- rename all class' members

There is more work to do but this pull request is getting too big so it is better to do this in chunks.

### CODE REVIEW

File: src/engine/localevent.cpp [hunk 1]
```diff
@@ -927,29 +940,29 @@ namespace EventProcessing
...
 #if defined( TARGET_NINTENDO_SWITCH )
             // Custom button mapping for Nintendo Switch
-            if ( buttonType == LocalEvent::CONTROLLER_BUTTON_A ) {
-                buttonType = LocalEvent::CONTROLLER_BUTTON_B;
+            if ( buttonType == LocalEvent::ControllerButtonType::CONTROLLER_BUTTON_A ) {
+                buttonType = LocalEvent::ControllerButtonType::CONTROLLER_BUTTON_B;
             }
-            if ( buttonType == LocalEvent::CONTROLLER_BUTTON_B ) {
-                buttonType = LocalEvent::CONTROLLER_BUTTON_A;
+            if ( buttonType == LocalEvent::ControllerButtonType::CONTROLLER_BUTTON_B ) {
```
Comment (Districh-ru) [inline src/engine/localevent.cpp:946]: Hi, @ihhub to fix Nintendo Switch gamepad buttons remap we should fix this line:
```suggestion
            else if ( buttonType == LocalEvent::ControllerButtonType::CONTROLLER_BUTTON_B ) {
```

Comment (ihhub) [reply src/engine/localevent.cpp]: Thanks for pointing out!

File: src/fheroes2/ai/normal/ai_normal_hero.cpp [hunk 2]
```diff
@@ -2463,6 +2463,8 @@ namespace AI
                 assert( tile.GetObject( false ) == MP2::OBJ_CASTLE && hero.GetColor() == Maps::getColorFromTile( tile ) );
                 assert( Maps::isValidDirection( tileIndex, Direction::BOTTOM ) && hero.GetIndex() == Maps::GetDirectionIndex( tileIndex, Direction::BOTTOM ) );
 
+                _enemyArmies.erase( tileIndex );
```
Comment (oleg-derevenetz) [inline src/fheroes2/ai/normal/ai_normal_hero.cpp:2466]: Hi @ihhub please fix the merge conflict by removing this line :)

**Review by oleg-derevenetz — CHANGES_REQUESTED** [review]

File: src/fheroes2/ai/normal/ai_normal_hero.cpp (hunk 2 above)
Comment (ihhub) [reply src/fheroes2/ai/normal/ai_normal_hero.cpp:2466]: Oops

**Review by oleg-derevenetz — APPROVED** [review]

**Review by Districh-ru — APPROVED** [review]

**Pull request merged**